lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -24,6 +24,15 @@ def fragment_cache(cached_hash, non_cached_hash)
         private
 
         def serializable_hash_for_collection(options)
+          if options[:batch_cache].blank? && ActiveModelSerializers.config.cache_store.present?
+            keys = CachedSerializer.object_cache_keys(serializer, @include_tree)
+            if keys.present?
+              values = ActiveModelSerializers.config.cache_store.read_multi(*keys)
+
+              options.merge!(batch_cache: values)
+            end
+          end
+
```

Comment:
I think this both blocks of code added in this diff should be their own method. Maybe something like (pardon the naming... could be better)

```ruby
def serializable_hash_for_collection(options)
  options.merge!(batch_cache(options))
  serializer.map { |s| Attributes.new(s, instance_options).serializable_hash(options) }
end
def batch_cache(options)
  return {} unless options[:batch_cache].blank? && ActiveModelSerializers.config.cache_store.present?
  if keys = CachedSerializer.object_cache_keys(serializer, @include_tree)
    { batch_cache: ActiveModelSerializers.config.cache_store.read_multi(*keys) }
  else
    {}
  end
end
def batch_cached_serializer(options)
  return nil unless options[:batch_cache].present?
  cache_key = CachedSerializer.new(serializer).cache_key
  options[:batch_cache][cache_key]
end
def resource_object_for(options)
  if batch_cached_serializer_attributes = batch_cached_serializer(options)
    batch_cached_serializer_attributes
  else
    cache_check(serializer) do
      serializer.attributes(options[:fields])
    end
end
```
  
  
  
def resource_object_for(options)